db/migrate/20170928202851_migrate_middleware_server_to_wildfly_and_eap.rb

```diff
@@ -0,0 +1,34 @@
+class MigrateMiddlewareServerToWildflyAndEap < ActiveRecord::Migration[5.0]
+  class MiddlewareServer < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    say_with_time('Migrating middleware_server to middleware_server_wildfly') do
+      MiddlewareServer
+        .where("type = ? AND product ~* ?", 'ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServer', 'wildfly')
+        .update_all(:type => 'ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServerWildfly')
```

Comment:
shouldn't the new wildfly/eap class be a subclass/subtype of middleware server class? here it seems it's in hierarchy of middleware_manager, or I am probably missing something

Reply:
I think they are better included in middleware_manager as well, so that we have proper namespacing for the resources, what do you think?

Another option would be `ManageIQ::Providers::Hawkular::WildFlyServer`. Any of the two are fine by me.